scripts/discourse-sync/client.py

```diff
@@ -0,0 +1,22 @@
+import pydiscourse
+
+# Client extends pydiscourse.DiscourseClient with new methods
+class Client(pydiscourse.DiscourseClient):
+    def lock_post(self, post_id, locked, **kwargs):
+        """
+        Lock a post from being edited
+        https://docs.discourse.org/#tag/Posts/operation/lockPost
+
+        Args:
+            post_id: integer
+            locked: bool
+        """
```

Comment:
This is even more of a nit, but a docstring should start with the summary line on the same line as the starting triple quote and then have a blank line between it and the rest of the content. If you're going to put the types in then they might as well be actual types. Something like this (this has a line from above so won't commit cleanly):

```suggestion
    def lock_post(self, post_id: int, locked: bool, **kwargs):
        """Lock a post from being edited.
        
        https://docs.discourse.org/#tag/Posts/operation/lockPost

        Args:
            post_id: the ID of the post to lock
            locked: True to lock, False to unlock
        """
```